node/http.go

```diff
@@ -0,0 +1,48 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+)
+
+type httpService struct {
+	srv      *http.Server
+	listener net.Listener
+	errCh    chan error
+}
+
+func (h *httpService) Start() error {
+	go func() {
+		if err := h.srv.Serve(h.listener); !errors.Is(err, http.ErrServerClosed) {
+			h.errCh <- fmt.Errorf("serve http: %v", err)
+		}
+	}()
+	return nil
+}
```

Comment:
No error is actually returned here, but instead is piped into `errCh`. Could simplify calling code via:

```suggestion
func (h *httpService) Start() {
	go func() {
		if err := h.srv.Serve(h.listener); !errors.Is(err, http.ErrServerClosed) {
			h.errCh <- fmt.Errorf("serve http: %v", err)
		}
	}()
}
```

Reply:
Nice. I modified the http service to use `Run(ctx context.Context) error` instead. I think it's clearer and it pushes concurrency up to the next level of abstraction.